diff --git a/es/CSSMotion.d.ts b/es/CSSMotion.d.ts
index d82f52aae39e88759f536b746a8e0efee2b0f8dd..ee3bd71e75518e5b5f3acc056f69647d5543a4e1 100644
--- a/es/CSSMotion.d.ts
+++ b/es/CSSMotion.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 import type { MotionEndEventHandler, MotionEventHandler, MotionPrepareEventHandler, MotionStatus } from './interface';
 export type CSSMotionConfig = boolean | {
     transitionSupport?: boolean;
diff --git a/es/CSSMotion.js b/es/CSSMotion.js
index c75a0519a5c7df267819f82fbfae824de4181f70..d1977b76fd9367ac2cc37171d3dda1f2379ad943 100644
--- a/es/CSSMotion.js
+++ b/es/CSSMotion.js
@@ -6,8 +6,8 @@ import _typeof from "@babel/runtime/helpers/esm/typeof";
 import classNames from 'classnames';
 import findDOMNode from "rc-util/es/Dom/findDOMNode";
 import { fillRef, supportRef } from "rc-util/es/ref";
-import * as React from 'react';
-import { useRef } from 'react';
+import * as React from 'preact/compat';
+import { useRef } from 'preact/compat';
 import { Context } from "./context";
 import DomWrapper from "./DomWrapper";
 import useStatus from "./hooks/useStatus";
diff --git a/es/CSSMotionList.d.ts b/es/CSSMotionList.d.ts
index 3f1ad5f946883d6830c9e698b1a9133e1e0aa530..3e001a84c93e011f032b6e7e3628320e124a6b66 100644
--- a/es/CSSMotionList.d.ts
+++ b/es/CSSMotionList.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 import type { CSSMotionProps } from './CSSMotion';
 import type { KeyObject } from './util/diff';
 export interface CSSMotionListProps extends Omit<CSSMotionProps, 'onVisibleChanged' | 'children'>, Omit<React.HTMLAttributes<any>, 'children'> {
diff --git a/es/CSSMotionList.js b/es/CSSMotionList.js
index 5c5da3c76b4b15a83f751e968836de1fa43e1185..db9467b195026fa8e9c30aafeb5982b6f785cf4e 100644
--- a/es/CSSMotionList.js
+++ b/es/CSSMotionList.js
@@ -10,7 +10,7 @@ import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
 var _excluded = ["component", "children", "onVisibleChanged", "onAllRemoved"],
   _excluded2 = ["status"];
 /* eslint react/prop-types: 0 */
-import * as React from 'react';
+import * as React from 'preact/compat';
 import OriginCSSMotion from "./CSSMotion";
 import { diffKeys, parseKeys, STATUS_ADD, STATUS_KEEP, STATUS_REMOVE, STATUS_REMOVED } from "./util/diff";
 import { supportTransition } from "./util/motion";
diff --git a/es/DomWrapper.d.ts b/es/DomWrapper.d.ts
index b3a43dc68615faed5366ca4dab471f44fdce5ed2..a96c3f8174bcb2c21e0f7c1cf27a1bbf50ccd2df 100644
--- a/es/DomWrapper.d.ts
+++ b/es/DomWrapper.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 export interface DomWrapperProps {
     children: React.ReactNode;
 }
diff --git a/es/DomWrapper.js b/es/DomWrapper.js
index a23fc9e9096a4b387a45d943410703d186017695..64f508b5f96329431adc49d6592d460ac1939837 100644
--- a/es/DomWrapper.js
+++ b/es/DomWrapper.js
@@ -2,7 +2,7 @@ import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
 import _createClass from "@babel/runtime/helpers/esm/createClass";
 import _inherits from "@babel/runtime/helpers/esm/inherits";
 import _createSuper from "@babel/runtime/helpers/esm/createSuper";
-import * as React from 'react';
+import * as React from 'preact/compat';
 var DomWrapper = /*#__PURE__*/function (_React$Component) {
   _inherits(DomWrapper, _React$Component);
   var _super = _createSuper(DomWrapper);
diff --git a/es/context.d.ts b/es/context.d.ts
index 006cdf691546991abde8976c1dd779d8000af4af..401ea75b5111339c66f02c789a5736d14c4000d1 100644
--- a/es/context.d.ts
+++ b/es/context.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 interface MotionContextProps {
     motion?: boolean;
 }
diff --git a/es/context.js b/es/context.js
index 38a2092568c08c472259bce6ab8011d386a9cc70..c266a2e73de3c7162c93d70d06acc20c95a89ddc 100644
--- a/es/context.js
+++ b/es/context.js
@@ -1,6 +1,6 @@
 import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
 var _excluded = ["children"];
-import * as React from 'react';
+import * as React from 'preact/compat';
 export var Context = /*#__PURE__*/React.createContext({});
 export default function MotionProvider(_ref) {
   var children = _ref.children,
diff --git a/es/hooks/useDomMotionEvents.js b/es/hooks/useDomMotionEvents.js
index a6fc092d6386913d4c610557b26f0678b2a3c9e2..d183f7adaa8ecb819205c8195fcf2c461c16d42c 100644
--- a/es/hooks/useDomMotionEvents.js
+++ b/es/hooks/useDomMotionEvents.js
@@ -1,5 +1,5 @@
-import * as React from 'react';
-import { useRef } from 'react';
+import * as React from 'preact/compat';
+import { useRef } from 'preact/compat';
 import { animationEndName, transitionEndName } from "../util/motion";
 export default (function (onInternalMotionEnd) {
   var cacheElementRef = useRef();
diff --git a/es/hooks/useIsomorphicLayoutEffect.d.ts b/es/hooks/useIsomorphicLayoutEffect.d.ts
index ac6065e2e3b75c166d82568f59cb78b61e317020..1f3ae0f1a03ce3c86c03f180ff7c0695bc155039 100644
--- a/es/hooks/useIsomorphicLayoutEffect.d.ts
+++ b/es/hooks/useIsomorphicLayoutEffect.d.ts
@@ -1,3 +1,3 @@
-import { useEffect } from 'react';
+import { useEffect } from 'preact/compat';
 declare const useIsomorphicLayoutEffect: typeof useEffect;
 export default useIsomorphicLayoutEffect;
diff --git a/es/hooks/useIsomorphicLayoutEffect.js b/es/hooks/useIsomorphicLayoutEffect.js
index 96dff4aac3ffccb8382f177afafaf4b5cf728091..dd044ad298cde926eaf52d90fb5d6a9223a5dcce 100644
--- a/es/hooks/useIsomorphicLayoutEffect.js
+++ b/es/hooks/useIsomorphicLayoutEffect.js
@@ -1,5 +1,5 @@
 import canUseDom from "rc-util/es/Dom/canUseDom";
-import { useEffect, useLayoutEffect } from 'react';
+import { useEffect, useLayoutEffect } from 'preact/compat';
 
 // It's safe to use `useLayoutEffect` but the warning is annoying
 var useIsomorphicLayoutEffect = canUseDom() ? useLayoutEffect : useEffect;
diff --git a/es/hooks/useNextFrame.js b/es/hooks/useNextFrame.js
index f1294055e05668a3fc6234bd3d2635f5a40839ef..4ad6a825e1c86e3fc6347e42cd623ccc866684f6 100644
--- a/es/hooks/useNextFrame.js
+++ b/es/hooks/useNextFrame.js
@@ -1,5 +1,5 @@
 import raf from "rc-util/es/raf";
-import * as React from 'react';
+import * as React from 'preact/compat';
 export default (function () {
   var nextFrameRef = React.useRef(null);
   function cancelNextFrame() {
diff --git a/es/hooks/useStatus.d.ts b/es/hooks/useStatus.d.ts
index 2059bd567ba564839fbba04da690c6cc9c2f8a04..56608648b29f30f737b915f9e127b7c28d5abce8 100644
--- a/es/hooks/useStatus.d.ts
+++ b/es/hooks/useStatus.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 import type { CSSMotionProps } from '../CSSMotion';
 import type { MotionStatus, StepStatus } from '../interface';
 export default function useStatus(supportMotion: boolean, visible: boolean, getElement: () => HTMLElement, { motionEnter, motionAppear, motionLeave, motionDeadline, motionLeaveImmediately, onAppearPrepare, onEnterPrepare, onLeavePrepare, onAppearStart, onEnterStart, onLeaveStart, onAppearActive, onEnterActive, onLeaveActive, onAppearEnd, onEnterEnd, onLeaveEnd, onVisibleChanged, }: CSSMotionProps): [MotionStatus, StepStatus, React.CSSProperties, boolean];
diff --git a/es/hooks/useStatus.js b/es/hooks/useStatus.js
index 3df964ac838898245c921398c7299ab206fb7743..3d4b0bf8b66e8882ca3ac59bc23494d1f2ebd75b 100644
--- a/es/hooks/useStatus.js
+++ b/es/hooks/useStatus.js
@@ -4,8 +4,8 @@ import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
 import { useEvent } from 'rc-util';
 import useState from "rc-util/es/hooks/useState";
 import useSyncState from "rc-util/es/hooks/useSyncState";
-import * as React from 'react';
-import { useEffect, useRef } from 'react';
+import * as React from 'preact/compat';
+import { useEffect, useRef } from 'preact/compat';
 import { STATUS_APPEAR, STATUS_ENTER, STATUS_LEAVE, STATUS_NONE, STEP_ACTIVE, STEP_PREPARE, STEP_PREPARED, STEP_START } from "../interface";
 import useDomMotionEvents from "./useDomMotionEvents";
 import useIsomorphicLayoutEffect from "./useIsomorphicLayoutEffect";
diff --git a/es/hooks/useStepQueue.js b/es/hooks/useStepQueue.js
index 9fad7246e1d0fa93bc8a8839ba467b9b8b89a57e..3d288adbf0506168709d074d4460a6703d78ff82 100644
--- a/es/hooks/useStepQueue.js
+++ b/es/hooks/useStepQueue.js
@@ -1,6 +1,6 @@
 import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
 import useState from "rc-util/es/hooks/useState";
-import * as React from 'react';
+import * as React from 'preact/compat';
 import { STEP_ACTIVATED, STEP_ACTIVE, STEP_NONE, STEP_PREPARE, STEP_PREPARED, STEP_START } from "../interface";
 import useIsomorphicLayoutEffect from "./useIsomorphicLayoutEffect";
 import useNextFrame from "./useNextFrame";
diff --git a/lib/CSSMotion.d.ts b/lib/CSSMotion.d.ts
index d82f52aae39e88759f536b746a8e0efee2b0f8dd..ee3bd71e75518e5b5f3acc056f69647d5543a4e1 100644
--- a/lib/CSSMotion.d.ts
+++ b/lib/CSSMotion.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 import type { MotionEndEventHandler, MotionEventHandler, MotionPrepareEventHandler, MotionStatus } from './interface';
 export type CSSMotionConfig = boolean | {
     transitionSupport?: boolean;
diff --git a/lib/CSSMotionList.d.ts b/lib/CSSMotionList.d.ts
index 3f1ad5f946883d6830c9e698b1a9133e1e0aa530..3e001a84c93e011f032b6e7e3628320e124a6b66 100644
--- a/lib/CSSMotionList.d.ts
+++ b/lib/CSSMotionList.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 import type { CSSMotionProps } from './CSSMotion';
 import type { KeyObject } from './util/diff';
 export interface CSSMotionListProps extends Omit<CSSMotionProps, 'onVisibleChanged' | 'children'>, Omit<React.HTMLAttributes<any>, 'children'> {
diff --git a/lib/DomWrapper.d.ts b/lib/DomWrapper.d.ts
index b3a43dc68615faed5366ca4dab471f44fdce5ed2..a96c3f8174bcb2c21e0f7c1cf27a1bbf50ccd2df 100644
--- a/lib/DomWrapper.d.ts
+++ b/lib/DomWrapper.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 export interface DomWrapperProps {
     children: React.ReactNode;
 }
diff --git a/lib/context.d.ts b/lib/context.d.ts
index 006cdf691546991abde8976c1dd779d8000af4af..401ea75b5111339c66f02c789a5736d14c4000d1 100644
--- a/lib/context.d.ts
+++ b/lib/context.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 interface MotionContextProps {
     motion?: boolean;
 }
diff --git a/lib/hooks/useIsomorphicLayoutEffect.d.ts b/lib/hooks/useIsomorphicLayoutEffect.d.ts
index ac6065e2e3b75c166d82568f59cb78b61e317020..1f3ae0f1a03ce3c86c03f180ff7c0695bc155039 100644
--- a/lib/hooks/useIsomorphicLayoutEffect.d.ts
+++ b/lib/hooks/useIsomorphicLayoutEffect.d.ts
@@ -1,3 +1,3 @@
-import { useEffect } from 'react';
+import { useEffect } from 'preact/compat';
 declare const useIsomorphicLayoutEffect: typeof useEffect;
 export default useIsomorphicLayoutEffect;
diff --git a/lib/hooks/useStatus.d.ts b/lib/hooks/useStatus.d.ts
index 2059bd567ba564839fbba04da690c6cc9c2f8a04..56608648b29f30f737b915f9e127b7c28d5abce8 100644
--- a/lib/hooks/useStatus.d.ts
+++ b/lib/hooks/useStatus.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 import type { CSSMotionProps } from '../CSSMotion';
 import type { MotionStatus, StepStatus } from '../interface';
 export default function useStatus(supportMotion: boolean, visible: boolean, getElement: () => HTMLElement, { motionEnter, motionAppear, motionLeave, motionDeadline, motionLeaveImmediately, onAppearPrepare, onEnterPrepare, onLeavePrepare, onAppearStart, onEnterStart, onLeaveStart, onAppearActive, onEnterActive, onLeaveActive, onAppearEnd, onEnterEnd, onLeaveEnd, onVisibleChanged, }: CSSMotionProps): [MotionStatus, StepStatus, React.CSSProperties, boolean];
diff --git a/package.json b/package.json
index 330d20f321c526782fc8e24ba42b0cd97b870341..115987373929d57dfc7594a6f6011f828c57f115 100644
--- a/package.json
+++ b/package.json
@@ -47,8 +47,8 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.11.1",
-    "classnames": "^2.2.1",
-    "rc-util": "^5.43.0"
+    "rc-util": "^5.43.0",
+    "clsx": "^2.1.1"
   },
   "devDependencies": {
     "@rc-component/father-plugin": "^1.0.1",
@@ -69,13 +69,11 @@
     "np": "^6.2.4",
     "prettier": "^2.1.1",
     "rc-test": "^7.0.14",
-    "react": "^18.3.0",
-    "react-dom": "^18.3.0",
-    "typescript": "^4.0.3"
+    "typescript": "^4.0.3",
+    "preact": "^10.22.1"
   },
   "peerDependencies": {
-    "react": ">=16.9.0",
-    "react-dom": ">=16.9.0"
+    "preact": "^10.22.1"
   },
   "cnpm": {
     "mode": "npm"
