diff --git a/es/Options.d.ts b/es/Options.d.ts
index cb957b28c8523eb0c4812b64ceef23e7696a6501..3a6a38b2e6383deeb66970b972ac6147845553a9 100644
--- a/es/Options.d.ts
+++ b/es/Options.d.ts
@@ -1,6 +1,6 @@
 import type { SelectProps } from 'rc-select';
 import type { OptionProps } from 'rc-select/es/Option';
-import React from 'react';
+import React from 'preact/compat';
 import type { PaginationLocale } from './interface';
 interface InternalSelectProps extends SelectProps {
     /**
diff --git a/es/Options.js b/es/Options.js
index 226b20a440313ee466daaff8c8297b9aba306303..ffdc6c35bd4aac25df0db3f6a7a70251f3e5dbc1 100644
--- a/es/Options.js
+++ b/es/Options.js
@@ -1,6 +1,6 @@
 import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
 import KEYCODE from "rc-util/es/KeyCode";
-import React from 'react';
+import React from 'preact/compat';
 var defaultPageSizeOptions = ['10', '20', '50', '100'];
 var Options = function Options(props) {
   var _props$pageSizeOption = props.pageSizeOptions,
diff --git a/es/Pager.d.ts b/es/Pager.d.ts
index 55c436c801f6002184d5c3ff3f3bfe2721a3bace..f7065d98165f715f498ac82efb328d4bfc3e809d 100644
--- a/es/Pager.d.ts
+++ b/es/Pager.d.ts
@@ -1,4 +1,4 @@
-import React from 'react';
+import React from 'preact/compat';
 import type { PaginationProps } from './interface';
 export interface PagerProps extends Pick<PaginationProps, 'itemRender'> {
     rootPrefixCls: string;
diff --git a/es/Pager.js b/es/Pager.js
index de3af5189d24633a8d93b4c3c2a8c121c9efc248..9c7321e7e8071d13e17bfa39bf16c1cd4a4006e1 100644
--- a/es/Pager.js
+++ b/es/Pager.js
@@ -1,7 +1,7 @@
 import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
 /* eslint react/prop-types: 0 */
 import classNames from 'classnames';
-import React from 'react';
+import React from 'preact/compat';
 var Pager = function Pager(props) {
   var _classNames;
   var rootPrefixCls = props.rootPrefixCls,
diff --git a/es/Pagination.d.ts b/es/Pagination.d.ts
index 7b09a3269a4421cc6e51a22dcb1eea59277d7ca9..c54983b05830c0c6e4cbc56fd4a34570618746cd 100644
--- a/es/Pagination.d.ts
+++ b/es/Pagination.d.ts
@@ -1,4 +1,4 @@
-import React from 'react';
+import React from 'preact/compat';
 import type { PaginationProps } from './interface';
 declare const Pagination: React.FC<PaginationProps>;
 export default Pagination;
diff --git a/es/Pagination.js b/es/Pagination.js
index 86848d8050fbd6b5df227f8dc5ad1e8320db6a81..661922b46b2368e14738475a5d73caf88eb54a1b 100644
--- a/es/Pagination.js
+++ b/es/Pagination.js
@@ -8,7 +8,7 @@ import useMergedState from "rc-util/es/hooks/useMergedState";
 import KeyCode from "rc-util/es/KeyCode";
 import pickAttrs from "rc-util/es/pickAttrs";
 import warning from "rc-util/es/warning";
-import React, { useEffect } from 'react';
+import React, { useEffect } from 'preact/compat';
 import zhCN from "./locale/zh_CN";
 import Options from "./Options";
 import Pager from "./Pager";
diff --git a/es/interface.d.ts b/es/interface.d.ts
index ab82180624baf1a0933b41aaffbf9a20d3819ce0..84028c5ae653087a6504cd5bd70517b69817dbe7 100644
--- a/es/interface.d.ts
+++ b/es/interface.d.ts
@@ -1,4 +1,4 @@
-import type React from 'react';
+import type React from 'preact/compat';
 export interface PaginationLocale {
     items_per_page?: string;
     jump_to?: string;
diff --git a/lib/Options.d.ts b/lib/Options.d.ts
index cb957b28c8523eb0c4812b64ceef23e7696a6501..3a6a38b2e6383deeb66970b972ac6147845553a9 100644
--- a/lib/Options.d.ts
+++ b/lib/Options.d.ts
@@ -1,6 +1,6 @@
 import type { SelectProps } from 'rc-select';
 import type { OptionProps } from 'rc-select/es/Option';
-import React from 'react';
+import React from 'preact/compat';
 import type { PaginationLocale } from './interface';
 interface InternalSelectProps extends SelectProps {
     /**
diff --git a/lib/Pager.d.ts b/lib/Pager.d.ts
index 55c436c801f6002184d5c3ff3f3bfe2721a3bace..f7065d98165f715f498ac82efb328d4bfc3e809d 100644
--- a/lib/Pager.d.ts
+++ b/lib/Pager.d.ts
@@ -1,4 +1,4 @@
-import React from 'react';
+import React from 'preact/compat';
 import type { PaginationProps } from './interface';
 export interface PagerProps extends Pick<PaginationProps, 'itemRender'> {
     rootPrefixCls: string;
diff --git a/lib/Pagination.d.ts b/lib/Pagination.d.ts
index 7b09a3269a4421cc6e51a22dcb1eea59277d7ca9..c54983b05830c0c6e4cbc56fd4a34570618746cd 100644
--- a/lib/Pagination.d.ts
+++ b/lib/Pagination.d.ts
@@ -1,4 +1,4 @@
-import React from 'react';
+import React from 'preact/compat';
 import type { PaginationProps } from './interface';
 declare const Pagination: React.FC<PaginationProps>;
 export default Pagination;
diff --git a/lib/interface.d.ts b/lib/interface.d.ts
index ab82180624baf1a0933b41aaffbf9a20d3819ce0..84028c5ae653087a6504cd5bd70517b69817dbe7 100644
--- a/lib/interface.d.ts
+++ b/lib/interface.d.ts
@@ -1,4 +1,4 @@
-import type React from 'react';
+import type React from 'preact/compat';
 export interface PaginationLocale {
     items_per_page?: string;
     jump_to?: string;
diff --git a/package.json b/package.json
index e253d0dd2347b8b4cf72fe97b99f5d1781d394ad..9636977955782d26ea43629f54d94d9c8af51827 100644
--- a/package.json
+++ b/package.json
@@ -43,8 +43,8 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.10.1",
-    "classnames": "^2.3.2",
-    "rc-util": "^5.38.0"
+    "rc-util": "^5.38.0",
+    "clsx": "^2.1.1"
   },
   "devDependencies": {
     "@rc-component/father-plugin": "^1.0.0",
@@ -71,12 +71,10 @@
     "prettier": "^3.1.0",
     "rc-select": "^14.1.2",
     "rc-test": "^7.0.15",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "preact": "^10.22.1"
   },
   "peerDependencies": {
-    "react": ">=16.9.0",
-    "react-dom": ">=16.9.0"
+    "preact": "^10.22.1"
   },
   "cnpm": {
     "mode": "npm"
