diff --git a/es/Popup.d.ts b/es/Popup.d.ts
index 6d78c360f36a2c185fa4e2a29e96bb6ea3ce776d..127b0faa6da8b07cf275a2480c055aabb58e1d2a 100644
--- a/es/Popup.d.ts
+++ b/es/Popup.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 export interface ContentProps {
     prefixCls?: string;
     children: (() => React.ReactNode) | React.ReactNode;
diff --git a/es/Popup.js b/es/Popup.js
index 9f0e0730b12ec5e682c2a65b79b68da454ffaf01..26cf8a3d112a8da0a9babcc3d3d29b7a95dfdf6d 100644
--- a/es/Popup.js
+++ b/es/Popup.js
@@ -1,5 +1,5 @@
 import classNames from 'classnames';
-import * as React from 'react';
+import * as React from 'preact/compat';
 export default function Popup(props) {
   var children = props.children,
     prefixCls = props.prefixCls,
diff --git a/es/Tooltip.d.ts b/es/Tooltip.d.ts
index a6b771062539e1a33c391b2313a911c4e89af7ca..917430d03d1a98ba76720f2f61b60478788bf266 100644
--- a/es/Tooltip.d.ts
+++ b/es/Tooltip.d.ts
@@ -1,6 +1,6 @@
 import type { ArrowType, TriggerProps } from '@rc-component/trigger';
 import type { ActionType, AlignType, AnimationType } from '@rc-component/trigger/lib/interface';
-import * as React from 'react';
+import * as React from 'preact/compat';
 export interface TooltipProps extends Pick<TriggerProps, 'onPopupAlign' | 'builtinPlacements' | 'fresh' | 'children' | 'mouseLeaveDelay' | 'mouseEnterDelay' | 'prefixCls' | 'forceRender'> {
     trigger?: ActionType | ActionType[];
     defaultVisible?: boolean;
diff --git a/es/Tooltip.js b/es/Tooltip.js
index daf1250371e78088edf6afc916fad97ad76ad09b..a055bdcf429ffe78b9793ef1b348c358d5443fca 100644
--- a/es/Tooltip.js
+++ b/es/Tooltip.js
@@ -3,7 +3,7 @@ import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
 import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
 var _excluded = ["overlayClassName", "trigger", "mouseEnterDelay", "mouseLeaveDelay", "overlayStyle", "prefixCls", "children", "onVisibleChange", "afterVisibleChange", "transitionName", "animation", "motion", "placement", "align", "destroyTooltipOnHide", "defaultVisible", "getTooltipContainer", "overlayInnerStyle", "arrowContent", "overlay", "id", "showArrow"];
 import Trigger from '@rc-component/trigger';
-import * as React from 'react';
+import * as React from 'preact/compat';
 import { forwardRef, useImperativeHandle, useRef } from 'react';
 import { placements } from "./placements";
 import Popup from "./Popup";
diff --git a/lib/Popup.d.ts b/lib/Popup.d.ts
index 6d78c360f36a2c185fa4e2a29e96bb6ea3ce776d..127b0faa6da8b07cf275a2480c055aabb58e1d2a 100644
--- a/lib/Popup.d.ts
+++ b/lib/Popup.d.ts
@@ -1,4 +1,4 @@
-import * as React from 'react';
+import * as React from 'preact/compat';
 export interface ContentProps {
     prefixCls?: string;
     children: (() => React.ReactNode) | React.ReactNode;
diff --git a/lib/Tooltip.d.ts b/lib/Tooltip.d.ts
index a6b771062539e1a33c391b2313a911c4e89af7ca..917430d03d1a98ba76720f2f61b60478788bf266 100644
--- a/lib/Tooltip.d.ts
+++ b/lib/Tooltip.d.ts
@@ -1,6 +1,6 @@
 import type { ArrowType, TriggerProps } from '@rc-component/trigger';
 import type { ActionType, AlignType, AnimationType } from '@rc-component/trigger/lib/interface';
-import * as React from 'react';
+import * as React from 'preact/compat';
 export interface TooltipProps extends Pick<TriggerProps, 'onPopupAlign' | 'builtinPlacements' | 'fresh' | 'children' | 'mouseLeaveDelay' | 'mouseEnterDelay' | 'prefixCls' | 'forceRender'> {
     trigger?: ActionType | ActionType[];
     defaultVisible?: boolean;
diff --git a/package.json b/package.json
index 0284808c92eca7b7d4bd8eb6ddb644b943040b44..7c6a5d5c2bcc756eb783bf68624a4a4c62554ed3 100644
--- a/package.json
+++ b/package.json
@@ -43,7 +43,7 @@
   "dependencies": {
     "@babel/runtime": "^7.11.2",
     "@rc-component/trigger": "^2.0.0",
-    "classnames": "^2.3.1"
+    "clsx": "^2.1.1"
   },
   "devDependencies": {
     "@rc-component/father-plugin": "^1.0.0",
@@ -60,9 +60,8 @@
     "less": "^4.1.1",
     "np": "^7.1.0",
     "rc-test": "^7.0.9",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
-    "typescript": "^4.0.5"
+    "typescript": "^4.0.5",
+    "preact": "^10.22.1"
   },
   "peerDependencies": {
     "react": ">=16.9.0",
